vault/plugin_catalog.go

```diff
@@ -531,77 +536,72 @@ func (c *PluginCatalog) Set(ctx context.Context, name string, pluginType consts.
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
-	// During plugin registration, we can't know if a plugin is multiplexed or
-	// not until we run it. So we set it to false here. Once started, we ask
-	// the plugin if it is multiplexed and set this value accordingly.
-	multiplexingSupport := false
-	return c.setInternal(ctx, name, pluginType, multiplexingSupport, command, args, env, sha256)
+	_, err := c.setInternal(ctx, name, pluginType, command, args, env, sha256)
+	return err
 }
 
-func (c *PluginCatalog) setInternal(ctx context.Context, name string, pluginType consts.PluginType, multiplexingSupport bool, command string, args []string, env []string, sha256 []byte) error {
+func (c *PluginCatalog) setInternal(ctx context.Context, name string, pluginType consts.PluginType, command string, args []string, env []string, sha256 []byte) (*pluginutil.PluginRunner, error) {
 	// Best effort check to make sure the command isn't breaking out of the
 	// configured plugin directory.
 	commandFull := filepath.Join(c.directory, command)
 	sym, err := filepath.EvalSymlinks(commandFull)
 	if err != nil {
-		return fmt.Errorf("error while validating the command path: %w", err)
+		return nil, fmt.Errorf("error while validating the command path: %w", err)
 	}
 	symAbs, err := filepath.Abs(filepath.Dir(sym))
 	if err != nil {
-		return fmt.Errorf("error while validating the command path: %w", err)
+		return nil, fmt.Errorf("error while validating the command path: %w", err)
 	}
 
 	if symAbs != c.directory {
-		return errors.New("cannot execute files outside of configured plugin directory")
+		return nil, errors.New("cannot execute files outside of configured plugin directory")
 	}
 
 	// If the plugin type is unknown, we want to attempt to determine the type
 	if pluginType == consts.PluginTypeUnknown {
 		// entryTmp should only be used for the below type check, it uses the
 		// full command instead of the relative command.
 		entryTmp := &pluginutil.PluginRunner{
-			Name:                name,
-			Command:             commandFull,
-			Args:                args,
-			Env:                 env,
-			Sha256:              sha256,
-			Builtin:             false,
-			MultiplexingSupport: multiplexingSupport,
+			Name:    name,
+			Command: commandFull,
+			Args:    args,
+			Env:     env,
+			Sha256:  sha256,
+			Builtin: false,
 		}
 
-		pluginType, multiplexingSupport, err = c.getPluginTypeFromUnknown(ctx, log.Default(), entryTmp)
+		pluginType, _, err = c.getPluginTypeFromUnknown(ctx, log.Default(), entryTmp)
```

Comment:
We can probably revert `getPluginTypeFromUnknown` to only return the pluginType and err. It doesn't appear to be needed to get multiplexingSupport. But maybe it could be useful in the future?

Reply:
Good call. Yeah we can revert this back, and the return values on `isDatabasePlugin` too.

Reply:
Updated in [bb267a7](https://github.com/hashicorp/vault/pull/14142/commits/bb267a7057ae9929d016c15716f41574ae9b9a8a).